contracts/verifiers/RequestOwnership.sol

```diff
@@ -50,9 +50,7 @@ abstract contract RequestOwnership is ZKPVerifierBase {
         uint64[] calldata requestIds,
         ZKPRequest[] calldata requests
     ) public virtual override {
-        for (uint256 i = 0; i < requestIds.length; i++) {
-            setZKPRequest(requestIds[i], requests[i]);
```

Comment:
We need to call `setZKPRequest` in RequestOwnership instead of `super.setZKPRequest` to call `_setRequestOwner` and set the request owner for these requests.
So this can't be optimized like this.